Cargo.toml

```diff
@@ -16,6 +16,8 @@ members = [
     "core/node/block_reverter",
     "core/node/commitment_generator",
     "core/node/house_keeper",
+    "core/node/genesis",
```

Comment:
Initially this PR introduced `zksync_node_shared` crate, but I changed it to be simply genesis.
Reasoning:
1. We have a tracked history of turning non-specific "shared stuff" crates into a mess, I don't really want us to have one more point of concentration for random stuff.
2. Genesis is an important part of the ZK Stack node, so it seems to be justified to have its own crate.